### PR TITLE
Add-exception-to-build_pp

### LINKED ISSUE #678: Add ImportError to build_powerplants.py for missing openjdk package.
## Describe the feature you'd like to see

The `build_powerplants` rule requires the openjdk package to work. However, if it is missing in the environment, the exception thrown does not make this clear. I would propose we add an [ImportError](https://docs.python.org/3/library/exceptions.html#ImportError) making this clear. The error arises here:

https://github.com/pypsa-meets-earth/pypsa-earth/blob/6a8871dafcc844703123377520669578999b86f1/scripts/build_powerplants.py#L298-L299

Or would it be better placed directly inside powerplantmatching?

### PR DESCRIPTION
**Pull request by Tomkourou**
# Closes #678.

## Changes proposed in this Pull Request

Hopefully this is the final one!

## Checklist

- [x] I consent to the release of this PR's code under the GPLv3 license and non-code contributions under CC0-1.0 and CC-BY-4.0.
- [ ] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Newly introduced dependencies are added to `envs/environment.yaml` and `envs/environment.docs.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml` and `config.tutorial.yaml`.
- [ ] Add a test config or line additions to `test/` (note tests are changing the config.tutorial.yaml)
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [x] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

File: scripts/build_powerplants.py [hunk 1]
```diff
@@ -294,6 +296,12 @@ def replace_natural_gas_fueltype(df):
         config["main_query"] = ppl_query
     else:
         config["main_query"] = ""
+    try:
+        check_output("java -version", stderr=STDOUT, shell=True).decode("utf-8")
+    except OSError:
+        logger.critical(
+            "java not found on path. You need to install openjdk to run build_powerplants.py"
+        )
```
Comment (davide-f) [inline scripts/build_powerplants.py]: Thank @Tomkourou!

I think we could encapsulate this into a function, like `def test_java_availability():` and place these lines as a content.

Then the call `test_java_availability()` can be placed  right at the beginning of main at around line 270.

I think we are close to finalize :)

**Review by davide-f — CHANGES_REQUESTED** [review]

**Comment by Tomkourou** [comment]
@davide-f do you want the function to be placed in `_helpers.py` ?

**Comment by davide-f** [comment]
> @davide-f do you want the function to be placed in `_helpers.py` ?

No need as it is a specific problem by powerplantmatching.
You may comment also about adding this feature there as well; they may be happy about it.

**Comment by davide-f** [comment]
Actually, someone noticed the same and proposed a similar fix
https://github.com/PyPSA/powerplantmatching/pull/109/files

Let's see wait the berlin team things, if it is merged into powerplantmatching, maybe it becomes redundant here

**Comment by davide-f** [comment]
I've just checked the PR into ppl and it seems that a similar fix will be available in ppl and it may be redundant here.
It may be worth if you could contribute directly there maybe proposing your approach, maybe first adding to the discussion (make a PR if they suggest them to do so).
What do you think?

**Comment by pz-max** [comment]
Closing as appears to be already solved in ppm https://github.com/PyPSA/powerplantmatching/pull/109/files

**Pull request closed without merging**